src/services/WebSocketPolyfill.js

```diff
@@ -112,7 +112,8 @@ export default function initWebSocketPolyfill(syncService, fileId, initialSessio
 				.findLast(s => s > 'AQ') || ''
 		}
 
-		close() {
+		async close() {
+			await this.#sendRemainingSteps()
```

Comment:
Great idea to try to send remaining steps before closing. I just wonder whether this might have user-facing implications on the responsibility of closing the session (e.g. when closing a document)?

Reply:
I tried and it was smooth... but that was with a fast network. Will try with throttling.

Reply:
Tried with slow 3g throtteling. The viewer closes fast and the first request in the background is the save request which writes the document. Then we push the remaining steps and update the awareness state and finally we send the close request. Due to the throtteling each request took 2 sec. - so after a total of 6 seconds the session was closed. However these were no noticable for the user as the viewer window closed fast - and it does not delay the saving of the document as that happens first.

So all in all i'd say that's the behavior we want to see. I'll test with collectives though as switching pages is a different scenario.

Reply:
Collectives also works great. The session `create` request for the next document happens in parallel with the `save` of the old. Then there's `push`, `sync` and `close`. Where `sync` is independent and happening in parallel.